### ISSUE TITLE
ReUnion and WinUI3 do not support taskbar control - TaskbarItemInfo

### ISSUE THREAD

**Issue by ivberg** [microsoft/microsoft-ui-xaml#5211]
**Describe the bug**
ReUnion and WinUI3 do not support taskbar control - TaskbarItemInfo. Both these projects promise to be additive and not take away from being able to use existing APIs or Win32 functionality. However, most popular win32 apps use taskbar controls like play/stop or other quick useful controls (introduced in Win7).

For C# - this is exposed via the TaskbarItemInfo class.
 https://docs.microsoft.com/en-us/dotnet/api/system.windows.shell.taskbariteminfo?view=net-5.0 

**Steps to reproduce the bug**
1. https://aka.ms/winui and per instructions create a C# Desktop WinUI3 0.5 app
2. Attempt to add taskbar functionality via https://docs.microsoft.com/en-us/dotnet/api/system.windows.shell.taskbariteminfo?view=net-5.0 
3. You will XAML compiler errors stating that for example <Window.CommandBindings> this error:
4. 
Error	WMC0001	Unknown type 'CommandBinding' in XML namespace 'http://schemas.microsoft.com/winfx/2006/xaml/presentation'	

**Expected behavior**
That ReUnion and WinUI3 support or allow these things just like WPF app would

**Screenshots**

**Version Info**
WinUI3, ReUnion 0.5.7

NuGet package version: 
PackageReference Include="Microsoft.ProjectReunion" Version="0.5.7"
PackageReference Include="Microsoft.ProjectReunion.Foundation" Version="0.5.7"
PackageReference Include="Microsoft.ProjectReunion.WinUI" Version="0.5.7"

Windows app type:
| UWP              | Win32            |
| :--------------- | :--------------- |
|  | Yes |

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (22000.40)              | Yes  |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |

**Additional context**

**Comment by asklar** [comment]
@ivberg you can add Reunion to an existing WPF app and call reunion APIs, but AFAIK the app must be WPF, not WinUI (WinUI doesn't know about WPF / WinForms apis).

**Comment by iceberg1370** [comment]
Yes seems to be the case. This would be a feature request to reimplement in WinUI with a helper class like TaskbarItemInfo similar to WPF. The idea for more WinUI adoption is to remove as many barriers to entry (or switching costs) as possible. Should be possible since these TaskBar items are native APIs underneath anyways.

**Comment by nikolayvpavlov** [comment]
Lack of integration with the taskbar is really disturbing. This is a selling point compared to web apps, and WinUI is missing it completely.

**Comment by nikolayvpavlov** [comment]
> You can use [ITaskbarList3 ](https://docs.microsoft.com/en-us/windows/win32/api/shobjidl_core/nn-shobjidl_core-itaskbarlist3)

This is the kind of attitude that alienates developers.  Taskbar support is there in WPF, and there is no need to go to interop.  Now, in WinUI, that is supposed to be the latest and greatest, we need to go down to bare metal for basic interaction with the Shell.  Please.  Why would one then want to use WinUI at all?

**Comment by castorix** [comment]
> ...Taskbar support is there in WPF...
> Why would one then want to use WinUI at all?

WinUI is based on UWP, and  they have probably higher priorities/bugs to fix
(and I don't see a lot of apps which use _ITaskbarList_* interfaces)

**Comment by nikolayvpavlov** [comment]
> (and I don't see a lot of apps which use _ITaskbarList_* interfaces)
I don't see a lot of apps which use UWP.  Even Microsoft own apps like Teams are based on Electron / WebView2 and are in essence. Not surprising with the death of the mobile platform and the strict sandboxing.

> ...  they have probably higher priorities/bugs to fix
I don't buy this either.  Both WinForms and WPF were very solid and quite feature-rich from v1.  Even UWP was in a decent shape, too.  WinUI starts as a hobbyist project, and the number of issues is ridiculous.  Looks like nothing from WPF / UWP is properly reused.

I just finished a small radio media player in WinUI and, I can say, I am far from impressed.  Too many quirks, too many weird visual effects, some controls failing to bind bi-directionally (RadioButtons) and limited to simple content (RadioButtons, NavigationView)...

And now you tell me I need to go to bare metal for proper integration with the Shell...  Not very encouraging.

**Comment by nikolayvpavlov** [comment]
Let alone that I am very curious how I can handle the WM_COMMAND message from WinUI.  Perhaps I am too dense, but I can't find a nice way to do it without going to interop and replacing the very window proc.  Huh.

**Comment by castorix** [comment]
> Let alone that I am very curious how I can handle the WM_COMMAND message from WinUI. 

With _SetWindowSubclass_ (like in the samples I posted)

A quick test with _ITaskbarList3_ and a Toolbar (like the MS SDK sample) : 

![WinUI3_ITaskbarList3_Test](https://user-images.githubusercontent.com/22345506/180427547-fde0fc7a-5393-4a34-b0c6-7b7db5a9bbab.gif)

**Comment by nikolayvpavlov** [comment]
@castorix , I honestly admire your work and effort, but there is no excuse that we need to go to interop for basic stuff like even command handling.  I see no reason why WndProc is not exposed for overriding like in WinForms.  Windows App Sdk is in its 1.1 version and feature-wise it is more of a 0.5.

**Comment by GuildOfCalamity** [comment]
I realize that MS is trying to decouple the window from the OS but it's a shame that these features were missed/avoided. 
It was so easy to add this to your **MainWindow**
```
    <Window.TaskbarItemInfo>
        <TaskbarItemInfo ProgressState="Normal" ProgressValue="{Binding BarProgress}" />
    </Window.TaskbarItemInfo>
```
And now you can't port any of your existing UWP/WPF window specific features to WinUI3.

**Comment by ryanlua** [comment]
@castorix Bit late to the party, but do you have the code you used for that example? I'd rather not struggle to guess what to interop for the thumbnail toolbar if it's already been done.

**Comment by castorix** [comment]
> [@castorix](https://github.com/castorix) Bit late to the party, but do you have the code you used for that example? I'd rather not struggle to guess what to interop for the thumbnail toolbar if it's already been done.

I did not upload it on my GitHub repository as it was a quick/draft code, but I put it on AppBox : https://app.box.com/s/0cl3eu48ixslwgdoks8jofq4zrs6buo3
